src/main/java/org/primefaces/component/fileupload/FileUploadRenderer.java

```diff
@@ -95,21 +87,27 @@ protected void encodeScript(FacesContext context, FileUpload fileUpload) throws
                     .callback("onAdd", "function(file, callback)", fileUpload.getOnAdd())
                     .callback("onstart", "function()", fileUpload.getOnstart())
                     .callback("onerror", "function()", fileUpload.getOnerror())
-                    .callback("onvalidationfailure", "function(msg)", fileUpload.getOnvalidationfailure())
                     .callback("oncancel", "function()", fileUpload.getOncancel())
                     .callback("oncomplete", "function(args)", fileUpload.getOncomplete());
 
-            String allowTypes = fileUpload.getAllowTypes();
-
-            if (allowTypes != null) {
-                wb.append(",allowTypes:").append(allowTypes);
-            }
         }
         else {
             wb.init("SimpleFileUpload", fileUpload.resolveWidgetVar(context), clientId)
-                    .attr("skinSimple", fileUpload.isSkinSimple(), false)
-                    .attr("maxFileSize", fileUpload.getSizeLimit(), Long.MAX_VALUE)
-                    .attr("invalidSizeMessage", EscapeUtils.forJavaScript(fileUpload.getInvalidSizeMessage()), null);
+                    .attr("skinSimple", fileUpload.isSkinSimple(), false);
+        }
+
+        wb.attr("disabled", fileUpload.isDisabled(), false)
+                .attr("invalidSizeMessage", fileUpload.getInvalidSizeMessage(), null)
```

Comment:
Why did you remove the `EscapeUtils` around invalidSize messsage?

Reply:
Hi @melloware 

The `EscapeUtils.forJavaScript()` call in `FileUploadRenderer` was removed due the following reasons:

1) It is already called by `WidgetBuilder.attr()`:
```java
public WidgetBuilder attr(String name, String value, String defaultValue) throws IOException {
        if (value != null && !value.equals(defaultValue)) {
            // ...
            rw.write(EscapeUtils.forJavaScript(value));
            // ...
        }
        return this;
    }
```
2) Calling `EscapeUtils.forJavaScript()` before the nullity check results in `"null"` strings instead of `null` references;
```java
assertEquals("null", EscapeUtils.forJavaScript(null)); 
```

I just added some tests for these cases.